src/light_data/light_store.rs

```diff
@@ -137,30 +141,49 @@ impl<F: LurkField> LightStore<F> {
     fn to_scalar_store(&self) -> anyhow::Result<ScalarStore<F>> {
         let mut store = ScalarStore::default();
         for (ptr, le) in self.scalar_map.iter() {
-            let se = match le {
-                None => None,
-                Some(LightExpr::Cons(x, y)) => Some(ScalarExpression::Cons(*x, *y)),
-                Some(LightExpr::Comm(f, x)) => Some(ScalarExpression::Comm(*f, *x)),
-                Some(LightExpr::Num(f)) => Some(ScalarExpression::Num(*f)),
-                Some(LightExpr::Char(f)) => {
+            let tag: ExprTag = ExprTag::from_field(ptr.tag()).ok_or(anyhow!("expr_tag"))?;
+            let se = match (tag, le) {
+                (ExprTag::Cons, Some(LightExpr::Cons(x, y))) => {
+                    Some(ScalarExpression::Cons(*x, *y))
+                }
+                (ExprTag::Comm, Some(LightExpr::Comm(f, x))) => {
+                    Some(ScalarExpression::Comm(*f, *x))
+                }
+                (ExprTag::Num, None) => Some(ScalarExpression::Num(*ptr.value())),
+                (ExprTag::Num, Some(LightExpr::Num(f))) => Some(ScalarExpression::Num(*f)),
```

Comment:
I'm still not sure I understand the range of `Num` instances the Light store is supposed to produce from this. When would we get one rather than the other?

Reply:
So, this is actually very mysterious to me too. I think in practice in the long run the right thing is to never produce either, and not have Num entries at all.

But for the moment, it seemed from my debugging of `intern_scalar_ptr` there were some cases where we were generating `None` entries on immediates, and this line (and the similar ones below) is part of eliminating that

Reply:
Okay, so I think we can resolve all these issues in the LightStore -> ScalarStore conversion by just saying that "temporarily we will accept a broader range of LightStores, such as those with nonsensical opaque immediate entries" (nonsensical because if its immediate, it can't be opaque by definition)

When we combine LightStore with ScalarStore, we should see if there's a way to better encode the properties of immediateness/opaqueness in the type system